README.md

```diff
@@ -1,57 +1,147 @@
+- should be unique within your Azure subscription.
+- `Azure Location`: The Azure location where your resources will be deployed.
+- `Azure Subscription`: The Azure Subscription where your resources will be deployed.
 
+> This template will create infrastructure and deploy code to Azure. If you don't have an Azure Subscription, you can sign up for
+> a [free account here](https://azure.microsoft.com/free/). Make sure you have contributor role to the Azure subscription.
 
-## Demo
+> NOTE: This may take a while to complete as it executes three commands: `azd init` (initializes environment), `azd provision`
+> (provisions Azure resources), and `azd deploy` (deploys application code). You will see a progress indicator as it provisions and
+> deploys your application.
 
-A demo app is included to show how to use the project.
+When `azd up` is complete it will output the following URLs:
 
-To run the demo, follow these steps:
+- Spring PetClinic Web application endpoint
+- Azure Portal link to view resources
 
-(Add steps to start up the demo)
+![azd up output](readme.assests/azd_up_output.png)
 
-1.
-2.
-3.
+You can also manage the created resources directly in `Azure Explorer` if you use `IntelliJ IDEA` and installed 
+[`Azure Toolkit for IntelliJ`](https://plugins.jetbrains.com/plugin/8053-azure-toolkit-for-intellij) for it
 
-## Resources
+![azure explorer](readme.assests/azure_explorer.png)
 
-(Any additional resources or related projects)
+Click the web application URL to launch the ToDo app. Create a new collection and add some items. This will create monitoring activity
+in the application that you will be able to see later when you run `azd monitor`.
 
-- Link to supporting information
-- Link to similar sample
-- ...
+> NOTE:
+>
+> - The `azd up` command will create Azure resources that will incur costs to your Azure subscription. You can clean up those resources
+> - manually via the Azure portal or with the `azd down` command.
+> - You can call `azd up` as many times as you like to both provision and deploy your solution, but you only need to provide the
+> - `--template` parameter the first time you call it to get the code locally. Subsequent `azd up` calls do not require the template
+> - parameter. If you do provide the parameter, all your local source code will be overwritten if you agree to overwrite when prompted.
+> - You can always create a new environment with `azd env new`.
+
+### Additional azd commands
+
+The Azure Developer CLI includes many other commands to help with your Azure development experience.
+You can view these commands at the terminal by running azd help. You can also view the full list of
+commands on our [Azure Developer CLI command](https://aka.ms/azure-dev/ref) page.
+
+## Run application locally
+
+To run this application locally with Azure solutions, you need to pass the environment variables below to the application first. Values of these
```

Comment:
there is such lib in java, but not applicable for azd .env, which is usually not be packaged:
1. We have no way to get the path of the `.env` file **at runtime**, because we have no way to reference `project.baseDir` at runtime, while the .env file only exists in the project and it won't be packaged into the final jar. 
2. This `.env` does not exist in Azure environment because it won't be packaged into the final jar, so it seems redundant to introduce such a lib (it will be packaged into the final jar).

Reply:
1. If using `VSCode`, we have the `dotenv` task that will load the env file.
2. If looking at pure shell scripting, there are some ways to accomplish this. You can use `azd env get-values` to return the current selected `.env` file values, and use some lightweight scripts to load it into the shell.

## Bash
```bash
export $(azd env get-values | xargs -0)
```

## PowerShell:
```pwsh
azd env get-values | % { "`$env:$_".Replace('\"', '`"') } | Invoke-Expression
```

Reply:
I added predefined tasks/configurations for VSCode users just same as other templates, and add more details about how to setup the environment variables in IntelliJ IDEA.

Reply:
I added predefined tasks/configurations for VSCode users leveraging the `dotenv` task just same as other templates and also added more details about how to setup environment variables in IntelliJ IDEA.